### ISSUE TITLE
Handling NAN values in NMF

### ISSUE THREAD

**Issue by PolarBean** [yoyololicon/pytorch-NMF#29]
There is this fantastic pull request in sklearn which does not apply the loss value to parts of the matrix which have nan values. https://github.com/scikit-learn/scikit-learn/pull/8474
Is this possible in pytorch-NMF at present?

**Comment by yoyolicoris** [comment]
Hi @PolarBean , thanks for asking.
I have a very brief look at the PR you mentioned.
It looks like it's a weighted version of NMF, and can be handled with masked tensors.
https://pytorch.org/tutorials/prototype/maskedtensor_overview
I'll have a proper look when I have the time, but it sounds doable.

**Comment by PolarBean** [comment]
very interesting, thank you!

**Comment by PolarBean** [comment]
After looking into it further it may not be possible in torch for a number of reasons. 

1. Neither dot or matmul are implemented for torch masked arrays. 
2. matmul and dot are not implemented for boolean datatypes. 

Im not particularly experienced with torch so there may be some workarounds, but this is the wall ive run up against!
